### PR TITLE
Convert databinding project to netstandard

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Should fix .NET Core issues when adding the nuget

**Note**: this will not fix runtime crashes. Those will be addressed in follow-up PR.

### CODE REVIEW

File: DataBinding/Realm.DataBinding.PCL/TypeInfoHelper.cs [hunk 1]
```diff
@@ -27,8 +27,7 @@ public static class TypeInfoHelper
         [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1600:ElementsMustBeDocumented")]
         public static TypeInfo GetInfo<T>() where T : RealmObject
         {
-            RealmPCLHelpers.ThrowProxyShouldNeverBeUsed();
-            return null;
+            return typeof(T).GetTypeInfo();
```
Comment (kristiandupont) [inline DataBinding/Realm.DataBinding.PCL/TypeInfoHelper.cs]: Why this change?

File: Realm/Realm/InteropConfig.cs [hunk 2]
```diff
@@ -38,18 +39,33 @@ internal static class InteropConfig
...
+            {
+            }
+
+            var assemblyLocationPI = typeof(Assembly).GetProperty("Location", BindingFlags.Public | BindingFlags.Instance);
+            if (assemblyLocationPI != null)
+            {
+                var assemblyLocation = Path.GetDirectoryName((string)assemblyLocationPI.GetValue(typeof(InteropConfig).GetTypeInfo().Assembly));
+                var folder = Path.Combine(assemblyLocation, "Documents");
+                Directory.CreateDirectory(folder);
+                return folder;
```
Comment (kristiandupont) [inline Realm/Realm/InteropConfig.cs]: So the folder will be `Documents` beneath the assembly location?

File: DataBinding/Realm.DataBinding.PCL/TypeInfoHelper.cs (hunk 1 above)
Comment (nirinchev) [inline DataBinding/Realm.DataBinding.PCL/TypeInfoHelper.cs]: I felt it's semantically incorrect to throw pcl exception and tell people to reference the platform specific version when they can't really do that (since we only have platform specific versions for iOS and Android). They shouldn't be using the library anyway, but if they somehow did, we can at least not throw :)

Comment (nirinchev) [inline DataBinding/Realm.DataBinding.PCL/TypeInfoHelper.cs]: To elaborate on that - our weaver will weave `IReflectableType` on pcl/net standard projects, because we can't know whether those will be used in an iOS application or a console app. While it's unlikely someone will be casting to `IReflectableType` in a console application, a 3rd party library may be doing it.

File: Realm/Realm/InteropConfig.cs (hunk 2 above)
Comment (nirinchev) [inline Realm/Realm/InteropConfig.cs]: Yes - it's far from ideal, which is why we should instruct users writing .NET Core apps to always specify the full path to the database.
